cmd/serve/s3/s3_test.go

```diff
@@ -200,5 +202,106 @@ func TestEncodingWithMinioClient(t *testing.T) {
 			}
 		})
 	}
+}
+
+type FileStuct struct {
+	path     string
+	filename string
+}
+
+var testdata = struct {
+	description string
+	bucket      string
+	files       []FileStuct
+}{
+	description: "list buckets",
+	bucket:      "mybucket",
+	files: []FileStuct{
+		{
+			path:     "",
+			filename: "lorem.txt",
+		},
+		{
+			path:     "foo",
+			filename: "bar.txt",
+		},
+	},
+}
+
+func testListBuckets(t *testing.T, useProxy bool) {
+	fstest.Initialise()
+
+	var f fs.Fs
+	if useProxy {
```

Comment:
I could not make this test to work in s3
https://github.com/rclone/rclone/blob/bf494d48d60765300d4c133ab8f6afc18a1089ad/cmd/serve/sftp/sftp_test.go#L39-L41

So instead I tried to add tests similar to 
https://github.com/rclone/rclone/blob/bf494d48d60765300d4c133ab8f6afc18a1089ad/cmd/serve/http/http_test.go#L92-L93